cli_tools/common/utils/daisy/daisy_utils.go

```diff
@@ -84,28 +85,44 @@ func UpdateAllInstanceNoExternalIP(workflow *daisy.Workflow, noExternalIP bool)
 	if !noExternalIP {
 		return
 	}
-	updateAllInstanceAccessConfig(workflow, func() []*compute.AccessConfig {
-		return []*compute.AccessConfig{}
-	})
-}
-
-func updateAllInstanceAccessConfig(workflow *daisy.Workflow, accessConfigProvider func() []*compute.AccessConfig) {
-	for _, step := range workflow.Steps {
-		if step.IncludeWorkflow != nil {
-			//recurse into included workflow
-			updateAllInstanceAccessConfig(step.IncludeWorkflow.Workflow, accessConfigProvider)
-		}
+	WorkflowStepIterator(workflow, func(step *daisy.Step) {
 		if step.CreateInstances != nil {
 			for _, instance := range *step.CreateInstances {
 				if instance.Instance.NetworkInterfaces == nil {
 					return
 				}
 				for _, networkInterface := range instance.Instance.NetworkInterfaces {
-					networkInterface.AccessConfigs = accessConfigProvider()
+					networkInterface.AccessConfigs = []*compute.AccessConfig{}
 				}
 			}
 		}
-	}
+	})
+}
+
+// UpdateToUEFICompatible updates workflow resources (disks and images) to be UEFI compatible
+func UpdateToUEFICompatible(workflow *daisy.Workflow) {
+	WorkflowStepIterator(workflow, func(step *daisy.Step) {
+		if step.CreateDisks != nil {
+			for _, disk := range *step.CreateDisks {
+				// for the time being, don't run Debian 9 worker in UEFI mode
+				if strings.Contains(disk.SourceImage, "debian-9-worker") {
+					continue
+				}
+				disk.Disk.GuestOsFeatures = daisy.CombineGuestOSFeatures(disk.Disk.GuestOsFeatures, "UEFI_COMPATIBLE")
```

Comment:
Some more background here would be good -- why are we not doing UEFI for Debian 9? What's blocking us? 

Also, can we encode this exception to Debian 9 elsewhere, outside of code, for example in the template/script that creates the Debian 9 worker? Two advantages: One, this current approach invites future one-off modifications, and two, that template/script will own all customizations regarding Debian 9.

Reply:
This is for Debian 9 worker, not for Debian 9 imports. Disks for workers are part of the workflow. I don't want to be changing how workers boot until UEFI is default.

Reply:
So how about moving this to a location that is specific to your worker?

In addition to the reasons above, here's another: If a user has the subtstring "debian-9-worker" in their source image's name and tries to use UEFI.

Reply:
Since import workflow only creates Debian 9 instances implicitly (no way for customer to have a CreateInstance with their own image in import workflow) the issue of customer provided image with "debian-9-worker" in it is not really an issue. Just in case, I switched to using the full URI to Debian 9 worker instead of just the portion.

What do you mean by "moving this to a location that is specific to your worker"?